ddsrouter_utils/include/ddsrouter_utils/utils.hpp

```diff
@@ -35,16 +35,29 @@ namespace eprosima {
 namespace ddsrouter {
 namespace utils {
 
-enum FileAccessMode
-{
-    EXIST       = 0,
-    READ        = 4,
-    WRITE       = 2,
+using FileAccessModeType = int;
 
-    // Windoes does not have execution info of a file
-    EXECUTION   = 1,
+enum class FileAccessMode
+{
+    exist               = 0,
+#if defined(_WIN32)
+    // See https://docs.microsoft.com/es-es/cpp/c-runtime-library/reference/access-waccess?view=msvc-170
+    read                = 4,
+    write               = 2,
+    read_write          = 6,
```

Comment:
Erasing `exec` for windows may make this code less multiplatform friendly. Why do you think this is better than showing a warning?